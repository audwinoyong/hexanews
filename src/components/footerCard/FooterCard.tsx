/**
 * Display footer card component.
 */
import * as React from 'react'
import Avatar from '@material-ui/core/Avatar'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'
import { History } from 'history'

import placeholder from '../../assets/placeholder.png'
import styles from './styles'

type Props = {
  title: string,
  history: History,
}

const FooterCard: React.StatelessComponent<Props> = ({
  title,
  history,
}) => {
  return (
    <div style={styles.root} onClick={() => history.push('/articleDetail')}>
      <Card style={styles.card}>
        <CardMedia
          style={styles.media}
          image={placeholder}
        />
        <CardContent style={styles.cardContent}>
          <Typography
            component='h2'
            gutterBottom
            variant='caption'
          >
            Related reads
          </Typography>
          <Typography
            component='h2'
            gutterBottom
            variant='subheading'
          >
            {title}
          </Typography>
          <div style={styles.profileContainer}>
            <Avatar style={styles.avatar}>
              EK
            </Avatar>
            <div style={styles.detailContainer as any}>
              <Typography
                variant='body1'
                style={styles.profileName}
              >
                {'Erik Kennedy'}
              </Typography>
              <Typography
                variant='body1'
                color='textSecondary'
                style={styles.date}
              >
                {'8 March 2018'}
              </Typography>
            </div>
          </div>
        </CardContent>
      </Card>
    </div>
  )
}

export default FooterCard