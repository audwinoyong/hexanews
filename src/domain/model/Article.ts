/**
 * Model for the article.
 */
export type Article = {
  title: string,
  description: string
  category?: string
  authorId?: string
}