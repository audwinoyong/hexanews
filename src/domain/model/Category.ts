/**
 * Model for category object in our application.
 */

export type Category = {
  _id?: string,
  title: string,
  description: string,
}