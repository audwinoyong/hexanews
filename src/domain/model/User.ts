/**
 * Model for the user.
 */
export type User = {
  _id: string,
  email: string,
  name: string,
  description: string,
  createdAt: string,
}