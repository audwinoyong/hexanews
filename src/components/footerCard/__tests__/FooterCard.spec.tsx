import * as React from 'react'
import * as renderer from 'react-test-renderer'
import * as Enzyme from 'enzyme'
import { shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

import FooterCard from '../FooterCard'
import { Article } from '../../../domain/model/Article'

Enzyme.configure({ adapter: new Adapter() })

const mockArticle: Article = {
  _id: '2',
  title: 'def',
  description: 'tester',
  category: 'tester',
  authorId: 'tester',
}

describe('Footer Card', () => {
  it('renders Footer Card', () => {
    const component = renderer.create(
      <FooterCard
        dispatch={jest.fn()}
        article={mockArticle}
        authorName='test'
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handle button go to article click', () => {
    const dispatch = jest.fn()
    const component = (
      <FooterCard
        dispatch={dispatch}
        article={mockArticle}
        authorName='test'
      />
    )
    const layout = shallow(component)
    layout.find('#footer-2').first().simulate('click')
    expect(dispatch).toHaveBeenCalled()
  })
})