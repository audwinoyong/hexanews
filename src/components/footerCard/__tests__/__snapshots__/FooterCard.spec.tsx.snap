// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Card renders Footer Card 1`] = `
<div
  id="footer-2"
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
      "flexGrow": 1,
    }
  }
>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
    style={
      Object {
        "boxShadow": "0px 0.25px 0.35px 0.35px rgba(0, 0, 0, 0.1)",
        "maxWidth": 385,
      }
    }
  >
    <div
      className="MuiCardMedia-root-29"
      src={undefined}
      style={
        Object {
          "height": 0,
          "paddingTop": "56.25%",
        }
      }
    />
    <div
      className="MuiCardContent-root-31"
      style={
        Object {
          "paddingRight": "2%",
        }
      }
    >
      <h2
        className="MuiTypography-root-32 MuiTypography-caption-42 MuiTypography-gutterBottom-49"
      >
        Related reads
      </h2>
      <h2
        className="MuiTypography-root-32 MuiTypography-subheading-39 MuiTypography-gutterBottom-49"
      >
        def
      </h2>
      <div
        style={
          Object {
            "display": "flex",
            "marginBottom": "3%",
          }
        }
      >
        <div
          className="MuiAvatar-root-57 MuiAvatar-colorDefault-58"
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "green",
              "height": 50,
              "width": 50,
            }
          }
        >
          T
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "marginLeft": 10,
            }
          }
        >
          <p
            className="MuiTypography-root-32 MuiTypography-body1-41"
            style={
              Object {
                "marginTop": 6,
              }
            }
          >
            test
          </p>
          <p
            className="MuiTypography-root-32 MuiTypography-body1-41 MuiTypography-colorTextSecondary-55"
            style={
              Object {
                "marginTop": 2,
              }
            }
          >
            21 September 2018
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
